File: leetcode/go/2938.separate-black-and-white-balls.go
func minimumSteps(s string) int64 {
    var ans, ones int64 = 0, 0
    for _, c := range s {
        if c == '1' {
            ones++
        } else {
            ans += ones
        }
    }
    return ans
}

/* */

func minimumSteps(s string) int64 {
    n := len(s)
    l, r := 0, n - 1
    var ans int64
    for l < r {
        if s[l] == '0' {
            l++
            continue
        }
        if s[r] == '1' {
            r--
            continue
        }
        ans += int64(r - l)
        r--
        l++
    }
    return ans
}

File: leetcode/rust-leetcode/src/2938.separate-black-and-white-balls.rs
impl Solution {
    pub fn minimum_steps(s: String) -> i64 {
        let (mut ones, mut ans) = (0,0);
        for b in s.bytes() {
            match b {
                b'1' => ones+=1,
                _ => ans += ones,
            }
        }
        ans
    }
}

